service/routes.py

```diff
@@ -28,14 +44,38 @@ def create_customers():
     """
     app.logger.info("Request to create a customer")
     check_content_type("application/json")
+
+    # initialize an empty Customer record
     customer = Customer()
+
+    # deserialize the request JSON into the newly created record
     customer.deserialize(request.get_json())
-    customer.create()
+
+    try:
+        # add the customer record to the database
+        customer.create()
+    except:
+        abort(
+            status.HTTP_400_BAD_REQUEST,
+            'Failed to create customer'
+        )
+
     message = customer.serialize()
-    location_url = url_for("get_customers", customer_id=customer.id, _external=True)
 
-    app.logger.info("Customer with ID [%s] created.", customer.id)
-    return jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
+    # TODO: this cannot be implemented until we have a functioning GET endpoint
+    # location_url = url_for("get_customers", customer_id=customer.id, _external=True)
```

Comment:
This is a tricky one: `url_for` builds URLs in one endpoint which point to another endpoint. In this case, POST endpoint is building a URL which points to the GET endpoint.

We should definitely resurrect this once we have GET done. But, until then, Flask doesn't like it when we try to build URLs to nonexistent endpoints